web-common/src/features/dashboards/scheduled-reports/time-utils.ts

```diff
@@ -0,0 +1,72 @@
+export function getTodaysDayOfWeek(): string {
+  const dayOfWeekMap: Record<number, string> = {
+    0: "Sunday",
+    1: "Monday",
+    2: "Tuesday",
+    3: "Wednesday",
+    4: "Thursday",
+    5: "Friday",
+    6: "Saturday",
+  };
+
+  return dayOfWeekMap[new Date().getDay()];
+}
+
+export function getNextQuarterHour(): Date {
+  const MS_PER_MINUTE = 60000;
+  const MINUTES_PER_QUARTER_HOUR = 15;
+  const MS_PER_QUARTER_HOUR = MS_PER_MINUTE * MINUTES_PER_QUARTER_HOUR;
+
+  const currentTime = new Date();
+  return new Date(
+    Math.ceil(currentTime.getTime() / MS_PER_QUARTER_HOUR) * MS_PER_QUARTER_HOUR
+  );
+}
```

Comment:
We have transformation methods present in `time` lib. Also luxon can be directly used as well